awswrangler/_data_types.py

```diff
@@ -467,7 +467,14 @@ def pyarrow_types_from_pandas(
     # Filling indexes
     indexes: List[str] = []
     if index is True:
-        for field in pa.Schema.from_pandas(df=df[[]], preserve_index=True):
+        try:
+            fields = pa.Schema.from_pandas(df=df[[]], preserve_index=True)
+        except AttributeError as ae:
+            if "'Index' object has no attribute 'head'" not in str(ae):
+                raise ae
+            # Get index fields from a new df with only index columns
+            fields = pa.Schema.from_pandas(df=df.reset_index().drop(columns=cols), preserve_index=False)
```

Comment:
Would it be better to simply use this line as the default instead of what is currently being used? It would avoid the try/except altogether

Side note, pa.Schema.from_pandas is the operation that is currently a bottleneck in my Ray S3 write work

Reply:
Fair enough - updated